components/apimgt/org.wso2.carbon.apimgt.rest.api.admin.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/admin/v1/impl/SettingsApiServiceImpl.java

```diff
@@ -92,4 +93,23 @@ public Response settingsScopesScopeGet(String username, String scopeName, Messag
         }
         return Response.ok().entity(scopeSettingsDTO).build();
     }
+
+    /**
+     * Get all scopes and respective roles
+     *
+     * @return Role-Scope list
+     */
+    @Override
+    public Response settingsScopesGet(MessageContext messageContext) {
+        try {
+            Map<String, String> roleScopeMapping = APIUtil.getRESTAPIScopesForTenant(MultitenantUtils
+                    .getTenantDomain(RestApiUtil.getLoggedInUsername()));
+            ScopeListDTO scopeListDTO = SettingsMappingUtil.fromScopeListToScopeListDTO(roleScopeMapping);
+            return Response.ok().entity(scopeListDTO).build();
+        } catch (Exception e) {
```

Comment:
Shall we catch the specific Exception here?

Reply:
PR has been reverted